pycheribuild/projects/syzkaller.py

```diff
@@ -37,117 +37,128 @@
+            GOROOT=goroot,
+            CC=self.commandline_to_str([self.CC, *self.essential_compiler_and_linker_flags]),
+            CXX=self.commandline_to_str([self.CXX, *self.essential_compiler_and_linker_flags]),
+            ADDCFLAGS=self.commandline_to_str(self.default_compiler_flags + self.default_ldflags),
+        )
+        cflags = self.default_compiler_flags + self.default_ldflags
+        self.make_args.set_env(CFLAGS=" ".join(cflags))
+        self.make_args.set_env(PATH=f'{goroot / "bin"}:{self.config.dollar_path_with_other_tools}')
 
     def syzkaller_install_path(self):
-        return self.config.cheri_sdk_bindir
+        return self.real_install_root_dir / "bin"
 
     def syzkaller_binary(self):
-        return self.config.cheri_sdk_bindir / "syz-manager"
+        return self.syzkaller_install_path() / "syz-manager"
 
     def needs_configure(self) -> bool:
         return False
 
     def compile(self, **kwargs):
-        cflags = self.default_compiler_flags + self.default_ldflags
-
-        self.make_args.set_env(
-            HOSTARCH="amd64",
-            TARGETARCH=self.crosscompile_target.cpu_architecture.value,
-            TARGETOS="freebsd",
-            GOROOT=self.goroot.expanduser(),
-            GOPATH=self.gopath.expanduser(),
-            CC=self.CC, CXX=self.CXX,
-            PATH=self._new_path)
         if self.sysgen:
             self.generate()
-
-        self.make_args.set_env(CFLAGS=" ".join(cflags))
-        self.run_make(parallel=False, cwd=self.gosrc)
+        self.run_make(parallel=False, cwd=self.source_dir)
 
     def generate(self):
-        with self.set_env(PATH=self._new_path, SOURCEDIR=self.cheribsd_dir):
-            self.run_make("extract", parallel=False, cwd=self.gosrc)
-            self.run_make("generate", parallel=False, cwd=self.gosrc)
+        cheribsd_target = self.crosscompile_target.get_rootfs_target()
+        cheribsd_dir = BuildCHERIBSD.get_source_dir(self, cross_target=cheribsd_target)
+        if not cheribsd_dir.exists():
+            self.dependency_error("Missing CheriBSD source directory")
+        with self.set_env(SOURCEDIR=cheribsd_dir):
+            self.run_make("extract", parallel=False, cwd=self.source_dir)
+            self.run_make("generate", parallel=False, cwd=self.source_dir)
 
     def install(self, **kwargs):
-        # XXX-AM: should have a propert install dir configuration
         native_build = self.source_dir / "bin"
-        mips64_build = native_build / "freebsd_mips64"
-        syz_remote_install = self.syzkaller_install_path() / "freebsd_mips64"
+        target_build = native_build / f"freebsd_{self.syz_arch}"
+        syz_remote_install = self.syzkaller_install_path() / f"freebsd_{self.syz_arch}"
 
         self.makedirs(syz_remote_install)
 
         self.install_file(native_build / "syz-manager", self.syzkaller_binary(), mode=0o755)
 
         if not self.config.pretend:
             # mips64_build does not exist if we preted, so skip
```

Comment:
Probably remove the stale comment or update it. `os.listdir` is also old, I'm not sure why I used it instead of pathlib, maybe this can be simplified.